tasks/EchoesToFrameConverterTask.cpp

```diff
@@ -0,0 +1,151 @@
+// documentation about them.
+
+bool EchoesToFrameConverterTask::configureHook()
+{
+    if (!EchoesToFrameConverterTaskBase::configureHook())
+        return false;
+    configureOutput();
+    updateLookUpTable();
+    return true;
+}
+bool EchoesToFrameConverterTask::startHook()
+{
+    if (!EchoesToFrameConverterTaskBase::startHook())
+        return false;
+    return true;
+    m_last_sample = base::Time::now();
+}
+void EchoesToFrameConverterTask::updateHook()
+{
+    EchoesToFrameConverterTaskBase::updateHook();
+
+    Radar radar_echo;
+    if (_echo.read(radar_echo, false) == RTT::NewData) {
+        base::Angle yaw_correction = base::Angle::fromRad(0);
+        base::samples::RigidBodyState sensor2ref_pose;
+        if (_sensor2ref_pose.read(sensor2ref_pose) != RTT::NoData) {
+            yaw_correction = base::Angle::fromRad(sensor2ref_pose.getYaw());
+        }
+
+        do {
+            if (m_current_sweep_size != radar_echo.sweep_length ||
+                m_current_num_angles != (2 * M_PI / radar_echo.step_angle.getRad()) ||
+                m_current_range != radar_echo.range) {
+                m_current_sweep_size = radar_echo.sweep_length;
+                m_current_num_angles = 2 * M_PI / radar_echo.step_angle.getRad();
+                m_current_range = radar_echo.range;
+                updateLookUpTable();
+            }
+            addEchoesToFrame(radar_echo, yaw_correction);
+        } while (_echo.read(radar_echo, false) == RTT::NewData);
+
+        if (base::Time::now() - m_last_sample >
+            _export_config.get().time_between_frames) {
+            publishFrame();
+        }
+    }
+}
+void EchoesToFrameConverterTask::errorHook()
+{
+    EchoesToFrameConverterTaskBase::errorHook();
+}
+void EchoesToFrameConverterTask::stopHook()
+{
+    EchoesToFrameConverterTaskBase::stopHook();
+}
+void EchoesToFrameConverterTask::cleanupHook()
+{
+    EchoesToFrameConverterTaskBase::cleanupHook();
+}
+
+void EchoesToFrameConverterTask::updateLookUpTable()
+{
+    LOG_INFO_S << "Updating LookUpTable";
+    auto config = _export_config.get();
+    m_lut.reset(new EchoToImageLUT(m_current_num_angles,
+        m_current_sweep_size,
+        config.beam_width,
+        config.output_image_size));
+    m_echoes = std::vector<uint8_t>(m_current_sweep_size * m_current_num_angles, 0);
+}
+
+void EchoesToFrameConverterTask::addEchoesToFrame(Radar const& echo,
+    base::Angle yaw_correction)
+{
+    LOG_INFO_S << "Adding echoes...";
+    double angle = (echo.start_heading + yaw_correction).getRad();
+    if (angle < 0) {
+        angle = 2 * M_PI + angle;
+    }
+
+    int start_angle_unit = round(angle / echo.step_angle.getRad());
+    int angles_in_a_frame = round(2 * M_PI / echo.step_angle.getRad());
+    for (int current_angle = 0;
+         current_angle < static_cast<int>(echo.sweep_timestamps.size());
+         current_angle++) {
+        std::copy(echo.sweep_data.begin() + current_angle * echo.sweep_length,
```

Comment:
formatting incosistency

Reply:
This is formatted according to our developers guide.

Reply:
Tbh, it should be formatted according to clang-format. Is it configured ? Has it formatted this file ?

(@jhonasiv that should really be the question when it comes to formatting ...)

Reply:
Sorry, thats how I usually mention such errors. But in this case the error was mine, I didnt realize that line 107 was still the for loop definition, so to me it looked as an clear identation difference. My bad.